app/controllers/invitations_controller.rb

```diff
@@ -9,23 +9,33 @@ class InvitationsController < Devise::InvitationsController
 
   def new
     authorize User
+    session[:invitation_params] = {}
```

Comment:
I'm interested in why we need a session cookie here. What would the implications be of saving the user and then updating with require 2sv?

I guess I feel like using a session cookie introduces some extra complexity, I'm not sure if it has some security and/or PII concerns, and might introduce an extra zendesk layer of debugging if a user doesn't allow cookies and runs into some weird behaviour. So I'd rather avoid unless there's a strong reason.

Reply:
The main reason for using a session cookie was so we didn't persist any data into the database after the first part of the form is submitted.  By saving to the database, the user would receive their invitation email after the first part of the form is completed.  If they are quick and do this before the second part of the form is submitted, they'll be able to set up their account without 2SV (although they'll be prompted next time they login).

I don't think it's possible to use Signon without having cookies enabled, so disabled cookies won't be an issue.

I'll have a look and see what would happen if I remove the cookie here and write directly to the database after the first step.

Reply:
Ah got you. Gah, this is kind of why I hate callbacks. Interested to see if you find a way around it.

Reply:
I've removed the need for cookies in [this commit](https://github.com/alphagov/signon/pull/2105/commits/0eb62f4e6ae890cb34c22da63602c425fb0df582).  I'd be keen to get your views on it.

Although I prefer the solution without cookies from a technical perspective, it means we end up sending users 2 emails (if they have 2SV mandated, but are from a non-2SV organisation): one to set up their account and the other telling them about 2SV.

Reply:
This is done, I've now removed the use of session cookies here.